scipy/ndimage/tests/test_filters.py

```diff
@@ -757,15 +755,14 @@ def test_correlate_complex_input_and_kernel(self, dtype, dtype_output, xp):
                             [1 + 4j, 5, 6j]], dtype=dtype)
         self._validate_complex(xp, array, kernel, dtype_output)
 
+    @xfail_xp_backends('cupy', reasons=["cupy/cupy#8405"])
+    @skip_xp_backends(np_only=True,
+                      reasons=["output=dtype is numpy-specific"],
+                      exceptions=['cupy'],)
```

Comment:
@ev-br what do you think about cases like this? Is this the nicest way to express what we want? I having a hard time thinking how we would condense this logic into one decorator rather than two, and whether that would even be preferable.

Reply:
Well, from readability POV this looks horrible TBH. Not that the original was pretty, but at least I can quickly see what's going on (maybe place `is_cupy` check first --- then the original reads "if cupy, there's a know cupy issue to fix in cupy, otherwise it's numpy and cupy only"). 

This version I simply cannot read without going back and forth and thinking for a while. This `skip_xp_backends(np_only=True` --- ok, so it's numpy only, what is "exceptions"? Ah, OK, it's probably allows cupy (got to go back and find the docs/implementation of skip_xp_backends, but let me guess for now), so cupy is an exception, but this was on the previous line, is it skipped or xfailed on neither? I'm confused and I surely won't be able to type this myself if I ever need something like this elsewhere without copy-pasting.
Unlike the previous version which is ugly but is possible to recreate.

ISTM thisis a good indicator that `skip_xp_backends` is already not very intuitive and overloading it further is not a very good idea. 

One thing to note is that `skip_xp_backends` operates on specific backend names, while regular pytest xfail and skip markers operate on predicates (and so does unittest.skipIf for that matter, and however the analog was called by nose --- so essentially every other testing framework).

So instead of overloading with `reasons=list` and `exceptions=list`, maybe there's a way to teach pytest to recognize `is cupy(xp) or not is_numpy(xp)` as a boolean.

Reply:
I agree that it is not intuitive, however `np_only=True, exceptions=['cupy']` is the best I have come up with so far implementation-wise to get `is cupy(xp) or not is_numpy(xp)`. Feel free to propose an alternative syntax and the necessary adjustments to the infrastructure!

Reply:
My comment could be condensed to "if you ask me, I'd leave it as it was until there's a tweak to the decorator infra;  but do roll with whatever best syntax you came up with if you feel strongly about it".